### PR TITLE
DAOS-4093 Integrate Java and Hadoop test into DAOS CI

### PR DESCRIPTION
**Pull request by zjf2012**
There are several changes here.

- install openjdk 8 and maven in docker files since we need them to build code. For ubuntu, we need to override default JDK, which is Java 11.
- set maven repository to google mirror due to the [issue](https://github.com/travis-ci/travis-ci/issues/6593). Otherwise, we may fail to download some dependencies.
- add new docker run in .travis.yml to build java.

### CODE REVIEW

**Comment by zjf2012** [comment]
> This seems to double the DAOS build time.
> 
> Is this just a test tool/kit? Does it dynamically link to the DAOS client? Could this be built once, outside of the DAOS build (i.e. in a separate project) and then the result of that build used in a PR/branch to just run the test to avoid the extra time to build in every PR? Like we do with a number of other test tools such as IOR, mdtest, the mpiio tests, etc.
> 
> Regardless of the answer to the above, should this test be run and report on Jenkins also?

It's not a integration test. It's just a build with unit-test which is memory operation and very quick. All of the should be completed within seconds. The time is spending on downloading dependencies. In our dev or travis machine, we can cache almost all dependencies which are released artifacts and seldom get changed.  But in docker, we need to build everything from scratch. Do we have our own docker registry? If yes, we can have an image with all dependencies ready. Then we don't need to download them repeatedly.

**Comment by johannlombardi** [comment]
@brianjmurrell To give you more context, the idea was to first integrate it into the travis build so that we have a better understanding of what is required before we also integrate it into our own Jenkins.

**Comment by brianjmurrell** [comment]
> @brianjmurrell To give you more context, the idea was to first integrate it into the travis build so that we have a better understanding of what is required before we also integrate it into our own Jenkins.

@johannlombardi Indeed, and a good plan of attack given that Travis builds don't impact our Jenkins pipeline and make a good testing ground.  I was just pointing out the effect it would have on our Jenkins builds if it were integrated as it is currently.

File: utils/docker/Dockerfile.ubuntu.18.04 [hunk 1]
```diff
@@ -28,7 +28,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
             libssl-dev libtool-bin libyaml-dev                          \
             locales make meson nasm ninja-build pandoc patch            \
             pylint python-dev python3-dev scons sg3-utils uuid-dev      \
-            yasm valgrind libhwloc-dev man python-distro
+            yasm valgrind libhwloc-dev man python-distro openjdk-8-jdk  \
+	    maven
```
Comment (johannlombardi) [inline utils/docker/Dockerfile.ubuntu.18.04:32]: This needs to be upgraded to 20.04, i am afraid now.

Comment (zjf2012) [reply utils/docker/Dockerfile.ubuntu.18.04]: done

**Pull request merged**